modules/repository-url/src/test/java/org/elasticsearch/repositories/url/URLFixture.java

```diff
@@ -40,18 +41,17 @@
     /**
      * Creates a {@link URLFixture}
      */
-    private URLFixture(final String workingDir, final String repositoryDir) {
-        super(workingDir);
+    private URLFixture(final int port, final String workingDir, final String repositoryDir) {
+        super(workingDir, port);
         this.repositoryDir = dir(repositoryDir);
     }
 
     public static void main(String[] args) throws Exception {
-        if (args == null || args.length != 2) {
-            throw new IllegalArgumentException("URLFixture <working directory> <repository directory>");
+        if (args == null || args.length != 3) {
+            throw new IllegalArgumentException("URLFixture <port> <working directory> <repository directory>");
         }
-
-        final URLFixture fixture = new URLFixture(args[0], args[1]);
-        fixture.listen();
+        final URLFixture fixture = new URLFixture(Integer.parseInt(args[0]), args[1], args[2]);
```

Comment:
Should we check that the directories actually exist and fail with a specific error if not? That would save some confusion later on if someone made a mistake in their test setup.

Reply:
makes sense. Added these additional checks